admin/server/alerts.go

```diff
@@ -73,9 +76,54 @@ func (s *Server) GetAlertMeta(ctx context.Context, req *adminv1.GetAlertMetaRequ
 		}
 	}
 
+	// Handle email recipients - create magic tokens for all recipients
+	recipientURLs := make(map[string]*adminv1.GetAlertMetaResponse_URLs)
+
+	var recipients []string
+	recipients = append(recipients, req.EmailRecipients...)
+	if req.AnonRecipients {
+		// add empty email for slack and other notifiers token
+		recipients = append(recipients, "")
+	}
+
+	if len(recipients) > 0 {
+		// Get owner email for comparison
+		var ownerEmail string
+		if req.OwnerId != "" {
+			owner, err := s.admin.DB.FindUser(ctx, req.OwnerId)
+			if err == nil {
+				ownerEmail = owner.Email
+			}
+		}
+
+		// Create magic tokens for all recipients
+		emailTokens, err := s.createMagicTokensAlert(ctx, proj.ID, req.Alert, req.OwnerId, recipients, attr)
+		if err != nil {
+			return nil, status.Errorf(codes.Internal, "failed to issue magic auth tokens: %s", err.Error())
+		}
```

Comment:
So it creates a new token for each report run? I actually thought for reports it re-used existing tokens with `FindReportTokensWithSecret`, but seems that's not the case. 

What do you think here? On the one hand, it's simpler to create new tokens every time, but can also end up with rather many tokens over time, even with expiration. But perhaps it's not really that bad. 

If you stick with creating every time, I guess we don't need `FindAlertTokens` and `FindAlertTokensWithSecret` (and similar for reports)?

Reply:
We were reusing earlier but then switched to creating new tokens everytime [here](https://github.com/rilldata/rill/pull/6216/files#diff-c2bcafed81fcbf8eb2837df22cb63d055009c3c64d5e9d354fbc41659e7b36aeL558). I also remember discussing this in past and thinking how many reports are currently being sent today and we decided to create on every run and expire after 6 months then changed it to expire every 3 months.